server/src/pages/shot/editor.js

```diff
@@ -30,12 +25,58 @@ exports.Editor = class Editor extends React.Component {
     this.onMouseUp = this.onMouseUp.bind(this);
     this.onMouseMove = this.onMouseMove.bind(this);
     this.selectedTool = React.createRef();
-    this.history = new EditorHistory(this.devicePixelRatio);
+  }
+
+  calculateCanvasPixelRatio() {
+    this.originalImage = new Image();
+    this.originalImage.crossOrigin = "Anonymous";
+    this.originalImage.onload = () => {
+      let canvasPixelRatio = 1;
+
+      if (this.props.clip.image.captureType !== "fullPage"
+          && this.props.clip.image.captureType !== "fullPageTruncated") {
+        canvasPixelRatio = this.originalImage.naturalWidth / this.props.clip.image.dimensions.x;
+      }
+
+      this.setState({canvasPixelRatio});
+      this.history = new EditorHistory(canvasPixelRatio);
+    };
+    this.originalImage.src = this.props.clip.image.url;
+  }
+
+  componentDidMount() {
+    this.calculateCanvasPixelRatio();
+    document.addEventListener("mouseup", this.onMouseUp);
+    this.setState({
+      tool: "pen",
+      color: "#000",
+      lineWidth: 5,
+      actionsDisabled: true
+    });
+  }
+
+  componentDidUpdate(oldProps, oldState) {
+    if ((!oldState.canvasPixelRatio && this.state.canvasPixelRatio) ||
+        oldState.resetCanvas !== this.state.resetCanvas) {
```

Comment:
These are the two places when the original shot image is rendered: the initial rendering once the pixel ratio is available, and when the reset button's pressed.